subprojects/gdk_rust/gdk_electrum/src/interface.rs

```diff
@@ -0,0 +1,946 @@
+                let script = p2shwpkh_script(&derived.public_key);
+                let blinding_key =
+                    asset_blinding_key_to_ec_private_key(&master_blinding_key, &script);
+                let public_key = ec_public_key_from_private_key(blinding_key);
+                let blinder = Some(public_key);
+                let addr = elements::Address::p2shwpkh(
+                    &derived.public_key,
+                    blinder,
+                    address_params(network),
+                );
+
+                Ok(BEAddress::Elements(addr))
+            }
+        }
+    }
+
+    pub fn get_settings(&self) -> Result<Settings, Error> {
+        Ok(self.db.get_settings()?.unwrap_or_default())
+    }
+
+    pub fn change_settings(&self, settings: &Settings) -> Result<(), Error> {
+        self.db.insert_settings(settings)
+    }
+
+    pub fn get_tip(&self) -> Result<u32, Error> {
+        self.db.get_tip()
+    }
+
+    pub fn list_tx(&self, opt: &GetTransactionsOpt) -> Result<Vec<TransactionMeta>, Error> {
+        let (_, all_txs) = self.db.get_all_spent_and_txs()?;
+        let all_scripts = self.db.get_all_scripts()?;
+        let all_unblinded = self.db.get_all_unblinded()?; // empty map if not liquid
+
+        let mut txs = vec![];
+        let mut my_txids = self.db.get_my()?;
+        my_txids.sort_by(|a, b| b.1.unwrap_or(std::u32::MAX).cmp(&a.1.unwrap_or(std::u32::MAX)));
+
+        for (tx_id, height) in my_txids.iter().skip(opt.first).take(opt.count) {
+            debug!("tx_id {}", tx_id);
+
+            let tx = all_txs.get(tx_id).ok_or_else(fn_err(&format!("list_tx no tx {}", tx_id)))?;
+            let header = height
+                .map(|h| self.db.get_header(h)?.ok_or_else(fn_err("no header")))
+                .transpose()?;
+
+            let mut addressees = vec![];
+            for i in 0..tx.output_len() as u32 {
+                let script = tx.output_script(i);
+                if !script.is_empty() && !all_scripts.contains(&script) {
+                    let address = tx.output_address(i, self.network.id());
+                    addressees.push(AddressAmount {
+                        address: address.unwrap_or_else(|| "".to_string()),
+                        satoshi: 0, // apparently not needed in list_tx addressees
+                        asset_tag: None,
+                    });
+                }
+            }
+            let create_transaction = CreateTransaction {
+                addressees,
+                ..Default::default()
+            };
+
+            let fee = tx.fee(&all_txs, &all_unblinded);
+            let satoshi = tx.my_balances(&all_txs, &all_scripts, &all_unblinded);
+
+            let negatives = satoshi.iter().filter(|(_, v)| **v < 0).count();
+            let positives = satoshi.iter().filter(|(_, v)| **v > 0).count();
+            let (type_, user_signed) =
+                match (positives > negatives, tx.is_redeposit(&all_scripts, &all_txs)) {
```

Comment:
added the following comment in eb7d165f0d78d460155aaf55c24aad611465f5aa
```
            // We define an incoming txs if there are more assets received by the wallet than spent
            // when they are equal it's an outgoing tx because the special asset liquid BTC
            // is negative due to the fee being paid
            // TODO how do we label issuance tx?
```